src/net/p2p/managernormal.h

```diff
@@ -163,18 +159,14 @@ namespace P2P {
       /**
        * Constructor.
        * @param hostIp The manager's host IP/address.
-       * @param rdpos Pointer to the rdPoS object.
        * @param options Pointer to the options singleton.
        * @param storage Pointer to the blockchain's storage.
        * @param state Pointer to the blockchain's state.
        */
       ManagerNormal(
-        const boost::asio::ip::address& hostIp, rdPoS& rdpos,
-        const Options& options, const Storage& storage,
-        State& state
-      ) : ManagerBase(hostIp, NodeType::NORMAL_NODE, 50, options),
-      rdpos_(rdpos), storage_(storage), state_(state)
-      {};
+        const boost::asio::ip::address& hostIp, const Options& options,
+        const Storage& storage, State& state
+      ) : ManagerBase(hostIp, NodeType::NORMAL_NODE, 50, options), storage_(storage), state_(state) {};
```

Comment:
Since we are refactoring this, I think is a good idea do have a static const int for the node capacity (static const int DEFAULT_NODE_CAPACITY = 50;)

Reply:
Done :ok_hand:

Reply:
Move it to Options instead, with default (50) one in Options::binaryDefaultOptions

Perhaps we could make these globals parameters a variables instead of being located only within the function itself?  :eyes: 

```cpp
  return {
    rootPath,
    DEFAULT_WEB3_CLIENT_VERSION,
    DEFAULT_VERSION,
    DEFAULT_CHAINID,
    DEFAULT_CHAIN_OWNER,
    DEFAULT_WS_PORT,
    DEFAULT_NODE_CAPACITY
    DEFAULT_HTTP_PORT,
    DEFAULT_EVENT_BLOCK_CAP,
    DEFAULT_LOG_CAP,
    DEFAULT_DISCOVERY_NODES,
    DEFAULT_GENESIS,
    DEFAULT_GENESIS_TIMESTAMP,
    DEFAULT_GENESIS_SIGNER,
    DEFAULT_GENESIS_BALANCE_LIST,
    DEFAULT_GENESIS_VALIDATORS
  };
  ```
  Maybe the vars should be declared on the optionsdefaults.cpp tho.
  
  
  For ManagerDiscovery you can use DEFAULT_NODE_CAPACITY * 10 :wink:

Reply:
Pushed a commit that moves the following vars to the Options singleton to be initialized by it (+ default values on AIO script and Options::binaryDefaults):

* `minDiscoveryConns` and `minNormalConns` (minimum required number of connections for each type of node on P2PManager - defaults to 11 for both)
* `maxDiscoveryConns` and `maxNormalConns` (maximum allowed number of connections for each type of node on P2PManager - defaults to 200 for Discovery and 50 for Normal)
* `minValidators` (minimum required Validators for block creation and signing on rdPoS - defaults to 4)

Diff should be able to explain it better than I can. We can adjust default values later if needed (I decided to leave them at their original values for now for consistence, now that everything's centered at Options it should be fairly easier to do so). Please have a look, if everything's OK with this then I guess we're ready to merge it.